cmd/tealdbg/debugger.go

```diff
@@ -181,7 +184,54 @@ func (s *session) Step() {
 	func() {
 		s.mu.Lock()
 		defer s.mu.Unlock()
-		s.debugConfig = debugConfig{BreakAtLine: stepBreak}
+		s.debugConfig = debugConfig{StepBreak: true}
+	}()
+
+	s.resume()
+}
+
+func (s *session) StepOver() {
+	// Get the first TEAL opcode in the line
+	currentOp := strings.Fields(s.lines[s.line.Load()])[0]
+
+	func() {
+		s.mu.Lock()
+		defer s.mu.Unlock()
+		// Step over a function call (callsub op).
+		// Set a breakpoint at the next line and resume until we reach our
+		// desired call stack height.
+		if currentOp == "callsub" {
+			err := s.setBreakpoint(s.line.Load() + 1)
+			if err != nil {
```

Comment:
@algochoi When an error happens, I'm wondering if it's appropriate to modify `s.debugConfig` _at all_?

* The PR guards against configuring `StepBreak` while allowing other state modification (e.g. `StepOutOver`).
* While I think I don't yet fully understand the implications of a partial state change, it feels error prone to me.  And I'm imagining a partial state update breaks `Debugger.Update` heuristics.  
* For the moment, I'm leaving aside consideration for if/how errors ought to be propagated to the client.  Though that question is in my mind too.

Reply:
An error would happen in `setBreakpoint()` if we try to set a breakpoint past the program. In that case, it sounds okay to me if we `StepBreak` instead. `Update` would be called after execution is resumed by calling `resume()` - it needs the message from the `acknowledge` channel to continue and sends a `notification` back to the evaluator to continue execution.

I don't think this particular error is notable for the client (but maybe I'm not thinking deep enough). Worst case, we are on the last line of the program and the user wanted to `StepOver`, thus triggering the error in `setBreakpoint`. In that case, setting the breakpoint properly is not critical to the user anyway. 

Let me know if you think I'm missing something.

Reply:
So I think this is not possible in the CDT debugger, and the error should never trigger in any of the control functions, but a meaningful place where `setBreakpoint()` would produce an error would be in `Debugger.setBreakpointByUrl`. If some front-end were to set a bogus breakpoint, then we should return the error. I don't think it would be possible to set a bogus breakpoint in any of the `Control` functions because we have tight control of the execution and line numbers. Unfortunately we still have to consume the error or our linter will yell at us.

Reply:
@algochoi I appreciate the detailed response here.  You've convinced me that the PR's behavior is correct.  Thanks for your patience - Resolving.